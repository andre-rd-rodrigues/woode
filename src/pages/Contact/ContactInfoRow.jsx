import React from "react";
import FeatherIcon from "feather-icons-react";
import styles from "./contact.module.scss";

const ContactInfoRow = ({ icon, text }) => {
  return (
    <div className={styles.contactInfoRow}>
      <FeatherIcon icon={icon} />
      <p>{text}</p>
    </div>
  );
};

export default ContactInfoRow;
