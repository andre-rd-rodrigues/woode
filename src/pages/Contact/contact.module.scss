@import "styles/mixins.scss";

.contact {
  & div[class="contact_first_layer_col col"] {
    align-items: center;
    padding: 0 30px;
  }
  & h1 {
    margin: 120px 0 100px 0;
    text-align: center;
    font-size: 3em;
  }
  & form {
    max-width: 700px;
    & textarea {
      display: block;
      margin: auto;
      width: 100%;
    }
    & input {
      @include input;
      border-bottom: 1px solid black;
      margin: 30px 0;
      width: 100%;
    }
    & button {
      display: block;
      margin-top: 20px;
      @include squared-button(black);
    }
  }
}
.contactInfoRow {
  display: flex;
  align-items: center;
  margin: 20px 0;
  & p {
    margin: 0 0 0 10px;
  }
}
