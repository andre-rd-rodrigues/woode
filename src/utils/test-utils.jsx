// test-utils.jsx
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { Provider } from "react-redux";
import { createStore } from "redux";

// Import your own reducer
import reducer from "store/rootReducer";
import activateNotification from "store/middleware/activateNotification";
import { MemoryRouter } from "react-router";

let store, initialState;

store = createStore(reducer, initialState);
initialState = {
  entities: {
    cart: {
      items: [],
      amount: 0,
      totalPrice: 0
    }
  },
  ui: {
    notifications: {
      topNotification: {
        active: false,
        message: ""
      },
      newsletterNotification: {
        active: false
      }
    }
  }
};

const ConnectedComponent = (children) => {
  return (
    <MemoryRouter>
      <Provider store={store}>{children}</Provider>
    </MemoryRouter>
  );
};

function render(ui, { initialState, ...renderOptions } = {}) {
  const Wrapper = ({ children }) => {
    return (
      <MemoryRouter>
        <Provider store={store}>{children}</Provider>
      </MemoryRouter>
    );
  };

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render, initialState, ConnectedComponent };
