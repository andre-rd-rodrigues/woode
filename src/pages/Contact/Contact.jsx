import React from "react";
import { Col, Container, Row } from "react-bootstrap";
import ContactInfoRow from "./ContactInfoRow";
import styles from "./contact.module.scss";

const Contact = () => {
  const handleSubmit = (e) => {
    e.preventDefault();
    alert("Submited!");
  };
  return (
    <div className={styles.contact}>
      <h1>Contact us</h1>
      <Container>
        <Row>
          <Col className="contact_first_layer_col">
            <form onSubmit={handleSubmit}>
              <textarea name="" id="" cols="30" rows="10" />
              <Row>
                <Col>
                  <input
                    required
                    aria-required
                    type="text"
                    placeholder="Your Name"
                  />
                </Col>
                <Col>
                  <input
                    required
                    aria-required
                    type="email"
                    placeholder="Your Email"
                  />
                </Col>
              </Row>
              <button type="submit">Submit</button>
            </form>
          </Col>
          <Col className="contact_first_layer_col">
            <h2>Send us a message</h2>
            <p>
              Lorem ipsum dolor sit amet, comp uting of ore et dolore ma antemo
              enim. Quam quisq ue id diam.
            </p>
            <div>
              <ContactInfoRow text="woode@example.com" icon="mail" />
              <ContactInfoRow
                text="MÃ¤ster Samuelsgatan 10A, Finland"
                icon="home"
              />
              <ContactInfoRow text="+ 668 66 448 6452 99" icon="headphones" />
            </div>
          </Col>
        </Row>
      </Container>
    </div>
  );
};

export default Contact;
