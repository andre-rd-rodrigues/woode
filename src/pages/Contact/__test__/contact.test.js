import React from "react";
import renderer from "react-test-renderer";
import { ConnectedComponent } from "utils/test-utils";
import Contact from "../Contact";

describe("Contact us", () => {
  it("should render correctly", () => {
    const wrapper = renderer.create(ConnectedComponent(<Contact />));

    expect(wrapper).toMatchSnapshot();
  });

  // it("should show notification when submit is successfull", async () => {
  //   const { getByTestId, getByPlaceholderText, getByText, findByTestId } =
  //     render(<Contact />);

  //   const textarea = getByTestId("contact_textarea");
  //   const nameInput = getByPlaceholderText(/Your Name/i);
  //   const emailInput = getByPlaceholderText(/Your Email/i);
  //   const submitButton = getByText(/Submit/i);

  //   userEvent.type(textarea, "Something");
  //   userEvent.type(nameInput, "My name");
  //   userEvent.type(emailInput, "email@example.com");
  //   userEvent.click(submitButton);

  //   expect(await findByTestId("top_notification")).toBeInTheDocument();
  // });
});
