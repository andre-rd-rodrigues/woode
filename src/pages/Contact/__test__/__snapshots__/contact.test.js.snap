// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact us should render correctly 1`] = `
<div
  className="contact"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <h1
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(-90px) translateZ(0)",
      }
    }
  >
    Contact us
  </h1>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="contact_first_layer_col col"
      >
        <form
          onSubmit={[Function]}
          style={
            Object {
              "opacity": 0,
              "transform": "translateY(-90px) translateZ(0)",
            }
          }
        >
          <textarea
            cols="30"
            data-testid="contact_textarea"
            name="contact_textarea"
            required={true}
            rows="10"
          />
          <div
            className="row"
          >
            <div
              className="col"
            >
              <input
                placeholder="Your Name"
                required={true}
                type="text"
              />
            </div>
            <div
              className="col"
            >
              <input
                placeholder="Your Email"
                required={true}
                type="email"
              />
            </div>
          </div>
          <button
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
      <div
        className="contact_first_layer_col col"
      >
        <h2
          style={
            Object {
              "opacity": 0,
              "transform": "translateY(-90px) translateZ(0)",
            }
          }
        >
          Send us a message
        </h2>
        <p
          style={
            Object {
              "opacity": 0,
              "transform": "translateY(-90px) translateZ(0)",
            }
          }
        >
          Lorem ipsum dolor sit amet, comp uting of ore et dolore ma antemo enim. Quam quisq ue id diam.
        </p>
        <div
          style={
            Object {
              "opacity": 0,
              "transform": "translateY(-90px) translateZ(0)",
            }
          }
        >
          <div
            className="contactInfoRow"
          >
            <svg
              className="feather feather-mail "
              fill="none"
              height={24}
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width={24}
            >
              <g
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z\\"></path><polyline points=\\"22,6 12,13 2,6\\"></polyline>",
                  }
                }
              />
            </svg>
            <p>
              woode@example.com
            </p>
          </div>
          <div
            className="contactInfoRow"
          >
            <svg
              className="feather feather-home "
              fill="none"
              height={24}
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width={24}
            >
              <g
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z\\"></path><polyline points=\\"9 22 9 12 15 12 15 22\\"></polyline>",
                  }
                }
              />
            </svg>
            <p>
              MÃ¤ster Samuelsgatan 10A, Finland
            </p>
          </div>
          <div
            className="contactInfoRow"
          >
            <svg
              className="feather feather-headphones "
              fill="none"
              height={24}
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width={24}
            >
              <g
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M3 18v-6a9 9 0 0 1 18 0v6\\"></path><path d=\\"M21 19a2 2 0 0 1-2 2h-1a2 2 0 0 1-2-2v-3a2 2 0 0 1 2-2h3zM3 19a2 2 0 0 0 2 2h1a2 2 0 0 0 2-2v-3a2 2 0 0 0-2-2H3z\\"></path>",
                  }
                }
              />
            </svg>
            <p>
              + 668 66 448 6452 99
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
